### PR TITLE
feat: upload command for cli

### PR DESCRIPTION
**Pull request by abhstabs**
Hi @ojaswa1942, Changes for the upload command. Please have a look. 

Thank you. 

cc: @pandurangpatil

### CODE REVIEW

**Comment by ojaswa1942** [comment]
@abhstabs Can you share a screenshot from `core` with the `upload` command? It will give a better understanding of requirements before review. 
I'd recommend creating a corresponding issue and adding the required information there.

**Review by ojaswa1942 — CHANGES_REQUESTED** [review]

File: pkg/docker/docker.go [hunk 1]
```diff
@@ -324,6 +323,7 @@ func RunImage(opts ...RunImageOption) error {
 
 	// Generate container configurations
 	containerConfig := getBaseContainerConfig(image)
+	containerConfig.Entrypoint = runOptions.entrypoint
```
[the remark this answers is not in the document]
Comment (abhstabs) [reply pkg/docker/docker.go:326]: I tried that case. Simply running the `scan` command works as expected. No changes for entrypoint where made in the scan file

File: cmd/upload.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	// run image with options
+	err = docker.RunImage(
+		docker.OptionWithLatestImage(false), // because we already pull the image for access-key (with pullImage parameter)
+		docker.OptionWithEntrypoint(entrypoint),
+		docker.OptionWithArgs(commandArgs),
+		docker.OptionWithAttachedOutput(),
+		docker.OptionWithSourceVolume(fileutils.GetAbsolutePath(repository)),
+		docker.OptionWithUserConfigVolume(config.AppConfig.UserConfigurationFilePath),
```
[the remark this answers is not in the document]
Comment (abhstabs) [reply cmd/upload.go:90]: Yes. We're reading the consent from config file and updating it in case of any change.

File: cmd/upload.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	}
+
+	// "always pass -ic: even when internal rules are ignored (-i)"
+	commandArgs := []string{
+		config.AppConfig.Container.SourceCodeVolumeDir,
+	}
+
+	entrypoint := []string{
+		config.AppConfig.Container.PrivadoCoreBinPath, "upload",
+	}
```
Comment (ojaswa1942) [inline cmd/upload.go:81]: Refactor comments: 
1. Can we switch this order (entrypoint first)? Will make it easier for new contributors to relate.
2. We can also remove comments which are not required anymore.
3. Also, check the indent, looks a little weird. VSCode should automatically do it on save if you have the default Go extensions installed & configured

**Review by ojaswa1942 — CHANGES_REQUESTED** [review]

File: cmd/upload.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	PostRun: func(cmd *cobra.Command, args []string) {
+		telemetryPostRun(nil)
+	},
+}
+
+func sync(cmd *cobra.Command, args []string) {
+	repository := args[0]
+	debug, _ := cmd.Flags().GetBool("debug")
+	// overwriteResults, _ := cmd.Flags().GetBool("overwrite")
+
```
[the remark this answers is not in the document]
Comment (abhstabs) [reply cmd/upload.go:58]: already have that in place in core.

File: cmd/upload.go (hunk 3 above)
Comment (abhstabs) [reply cmd/upload.go:81]: Indent was VSCode automatic. I'll change the ordering and comments.

File: cmd/upload.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (ojaswa1942) [reply cmd/upload.go:58]: CLI needs to handle it independently as well.

**Review by ojaswa1942 — APPROVED** [review]
:shipit:

**Pull request merged**